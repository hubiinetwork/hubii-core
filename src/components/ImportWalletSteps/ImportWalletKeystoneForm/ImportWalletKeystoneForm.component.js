import * as React from 'react';
import { Form } from 'antd';
import PropTypes from 'prop-types';

import { ModalFormInput, ModalFormItem } from 'components/ui/Modal';
import { handleFinish } from 'utils/forms';
import fs from 'fs';
import { remote } from 'electron';

import {
  WidthEighty,
  StyledModalFormLabel,
  ButtonDiv,
  StyledBackButton,
  StyledButton,
  StyledSpan,
  FinalHeader,
} from '../ImportWalletForm.style';

class ImportWalletKeystoneForm extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      keystone: null,
      filepath: null,
    };
    this.loadKeystoneFile = this.loadKeystoneFile.bind(this);
  }

  loadKeystoneFile() {
    const { dialog } = remote;
    dialog.showOpenDialog((filenames) => {
      if (!filenames || !filenames.length) {
        this.setState({ keystone: null, filepath: null });
        return;
      }
      const encryptedWallet = fs.readFileSync(filenames[0], 'utf8');
      this.setState({ keystone: encryptedWallet, filepath: filenames[0] });
    });
  }

  render() {
    const { getFieldDecorator } = this.props.form;
    const { form, handleNext, loading } = this.props;
    const { keystone, filepath } = this.state;
    return (
      <div>
        <Form
          onSubmit={(e) => handleFinish(e, form, handleNext, { keystone })}
          layout="vertical"
          style={{
            marginTop: '2rem',
            display: 'flex',
            flexDirection: 'column',
            alignItems: 'center',
          }}
        >
          <WidthEighty>
            <FinalHeader>
              <p>Importing from a keystone file</p>
            </FinalHeader>
            <ModalFormItem
              label={
                <StyledModalFormLabel>
                  Enter a name for your wallet
                </StyledModalFormLabel>
              }
            >
              {getFieldDecorator('name', {
                rules: [
                  {
                    message: 'Please enter a name for your wallet',
                    required: true,
                    whitespace: true,
                  },
                ],
              })(<ModalFormInput disabled={loading} />)}
            </ModalFormItem>
            <ModalFormItem
              label={
                <StyledModalFormLabel>
                  {filepath}
                </StyledModalFormLabel>
              }
            >
              <StyledButton onClick={this.loadKeystoneFile}>
                <StyledSpan>Keystone File</StyledSpan>
              </StyledButton>
            </ModalFormItem>
            <ButtonDiv>
              <StyledBackButton type="default" onClick={this.props.handleBack}>
                <StyledSpan>Back</StyledSpan>
              </StyledBackButton>
              <StyledButton type="primary" disabled={!this.state.keystone} htmlType="submit">
                <StyledSpan>Import wallet</StyledSpan>
              </StyledButton>
            </ButtonDiv>
          </WidthEighty>
        </Form>
      </div>
    );
  }
}

ImportWalletKeystoneForm.propTypes = {
  /**
   * Function to be executed when back button is pressed
   */
  handleBack: PropTypes.func.isRequired,
  /**
   * Function to be executed when next is clicked.
   */
  handleNext: PropTypes.func.isRequired,
    /**
   * ant design form
   */
  form: PropTypes.object,
  /**
   * loading
   */
  loading: PropTypes.bool.isRequired,
};

export default Form.create()(ImportWalletKeystoneForm);
