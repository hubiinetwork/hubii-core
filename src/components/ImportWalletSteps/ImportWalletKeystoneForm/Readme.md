Variants of ImportWalletKeystoneForm:

```js
<div
  style={{
    backgroundColor: '#2f4d5c',
    padding: 25,
    width: 576,
    height: 576
  }}
>
  <ImportWalletKeystoneForm
    handleNext={() => console.log("handleNext")}
    handleNext={() => console.log("handleBack")}
    loading={false}
  />
</div>
```

```js
<div
  style={{
    backgroundColor: '#2f4d5c',
    padding: 25,
    width: 576,
    height: 576
  }}
>
  <ImportWalletKeystoneForm
    handleNext={() => console.log("handleNext")}
    handleNext={() => console.log("handleBack")}
    loading
  />
</div>
```