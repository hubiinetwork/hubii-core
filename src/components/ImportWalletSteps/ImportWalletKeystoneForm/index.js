import ImportWalletKeystoneForm from './ImportWalletKeystoneForm.component';
export default ImportWalletKeystoneForm;
