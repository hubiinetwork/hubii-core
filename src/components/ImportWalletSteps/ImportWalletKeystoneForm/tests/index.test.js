import React from 'react';
import { shallow } from 'enzyme';
import ImportWalletKeystoneForm from '../index';

describe('<ImportWalletKeystoneForm />', () => {
  it('should render correctly when loading is false', () => {
    const wrapper = shallow(<ImportWalletKeystoneForm loading={false} handleBack={() => {}} handleNext={() => {}} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should render correctly when loading is true', () => {
    const wrapper = shallow(<ImportWalletKeystoneForm loading handleBack={() => {}} handleNext={() => {}} />);
    expect(wrapper).toMatchSnapshot();
  });
});
