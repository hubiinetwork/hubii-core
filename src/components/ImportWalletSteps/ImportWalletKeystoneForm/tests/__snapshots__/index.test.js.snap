// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ImportWalletKeystoneForm /> should render correctly when loading is false 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Form(ImportWalletKeystoneForm)
    handleBack={[Function]}
    handleNext={[Function]}
    loading={false}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "form": Object {
        "getFieldDecorator": [Function],
        "getFieldError": [Function],
        "getFieldInstance": [Function],
        "getFieldProps": [Function],
        "getFieldValue": [Function],
        "getFieldsError": [Function],
        "getFieldsValue": [Function],
        "isFieldTouched": [Function],
        "isFieldValidating": [Function],
        "isFieldsTouched": [Function],
        "isFieldsValidating": [Function],
        "isSubmitting": [Function],
        "resetFields": [Function],
        "setFields": [Function],
        "setFieldsInitialValue": [Function],
        "setFieldsValue": [Function],
        "submit": [Function],
        "validateFields": [Function],
        "validateFieldsAndScroll": [Function],
      },
      "handleBack": [Function],
      "handleNext": [Function],
      "loading": false,
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "form": Object {
          "getFieldDecorator": [Function],
          "getFieldError": [Function],
          "getFieldInstance": [Function],
          "getFieldProps": [Function],
          "getFieldValue": [Function],
          "getFieldsError": [Function],
          "getFieldsValue": [Function],
          "isFieldTouched": [Function],
          "isFieldValidating": [Function],
          "isFieldsTouched": [Function],
          "isFieldsValidating": [Function],
          "isSubmitting": [Function],
          "resetFields": [Function],
          "setFields": [Function],
          "setFieldsInitialValue": [Function],
          "setFieldsValue": [Function],
          "submit": [Function],
          "validateFields": [Function],
          "validateFieldsAndScroll": [Function],
        },
        "handleBack": [Function],
        "handleNext": [Function],
        "loading": false,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<ImportWalletKeystoneForm /> should render correctly when loading is true 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Form(ImportWalletKeystoneForm)
    handleBack={[Function]}
    handleNext={[Function]}
    loading={true}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "form": Object {
        "getFieldDecorator": [Function],
        "getFieldError": [Function],
        "getFieldInstance": [Function],
        "getFieldProps": [Function],
        "getFieldValue": [Function],
        "getFieldsError": [Function],
        "getFieldsValue": [Function],
        "isFieldTouched": [Function],
        "isFieldValidating": [Function],
        "isFieldsTouched": [Function],
        "isFieldsValidating": [Function],
        "isSubmitting": [Function],
        "resetFields": [Function],
        "setFields": [Function],
        "setFieldsInitialValue": [Function],
        "setFieldsValue": [Function],
        "submit": [Function],
        "validateFields": [Function],
        "validateFieldsAndScroll": [Function],
      },
      "handleBack": [Function],
      "handleNext": [Function],
      "loading": true,
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "form": Object {
          "getFieldDecorator": [Function],
          "getFieldError": [Function],
          "getFieldInstance": [Function],
          "getFieldProps": [Function],
          "getFieldValue": [Function],
          "getFieldsError": [Function],
          "getFieldsValue": [Function],
          "isFieldTouched": [Function],
          "isFieldValidating": [Function],
          "isFieldsTouched": [Function],
          "isFieldsValidating": [Function],
          "isSubmitting": [Function],
          "resetFields": [Function],
          "setFields": [Function],
          "setFieldsInitialValue": [Function],
          "setFieldsValue": [Function],
          "submit": [Function],
          "validateFields": [Function],
          "validateFieldsAndScroll": [Function],
        },
        "handleBack": [Function],
        "handleNext": [Function],
        "loading": true,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
